File: rust/src/leetcode/Q136_single_number.rs
/*
 * @lc app=leetcode id=136 lang=rust
 *
 * [136] Single Number
 */

// @lc code=start
impl Solution {
    pub fn single_number(nums: Vec<i32>) -> i32 {
        let mut result: i32 = 0;

        for n in nums {
            result ^= n;
        }

        result
    }
}
// @lc code=end

File: python/Q136_single_number.py
#
# @lc app=leetcode id=136 lang=python3
#
# [136] Single Number
#


# @lc code=start
class Solution:
    def singleNumber(self, nums: list[int]) -> int:
        result: int = 0

        for n in nums:
            result ^= n

        return result


# @lc code=end
